tests/contrib/botocore/test.py

```diff
@@ -3765,3 +3774,414 @@ def test_schematized_unspecified_service_secretsmanager_v1(self):
+    @mock_sns
+    @mock_sqs
+    def test_aws_payload_tagging_sns(self):
+        with self.override_config("botocore", dict(payload_tagging_request="all", payload_tagging_response="all")):
+            region = "us-east-1"
+            sns = self.session.create_client("sns", region_name=region, endpoint_url="http://localhost:4566")
+
+            topic = sns.create_topic(Name="testTopic")
+
+            topic_arn = topic["TopicArn"]
+            sqs_url = self.sqs_test_queue["QueueUrl"]
+            url_parts = sqs_url.split("/")
+            sqs_arn = "arn:aws:sqs:{}:{}:{}".format(region, url_parts[-2], url_parts[-1])
+            sns.subscribe(TopicArn=topic_arn, Protocol="sqs", Endpoint=sqs_arn)
+
+            Pin(service=self.TEST_SERVICE, tracer=self.tracer).onto(sns)
+
+            message_attributes = {
+                "one": {"DataType": "String", "StringValue": "one"},
+                "two": {"DataType": "String", "StringValue": "two"},
+                "three": {"DataType": "String", "StringValue": "three"},
+                "four": {"DataType": "String", "StringValue": "four"},
+                "five": {"DataType": "String", "StringValue": "five"},
+                "six": {"DataType": "String", "StringValue": "six"},
+                "seven": {"DataType": "String", "StringValue": "seven"},
+                "eight": {"DataType": "String", "StringValue": "eight"},
+                "nine": {"DataType": "String", "StringValue": "nine"},
+                "ten": {"DataType": "String", "StringValue": "ten"},
+            }
+            entries = [
+                {"Id": "1", "Message": "ironmaiden", "MessageAttributes": message_attributes},
+                {"Id": "2", "Message": "megadeth", "MessageAttributes": message_attributes},
+            ]
+            sns.publish_batch(TopicArn=topic_arn, PublishBatchRequestEntries=entries)
+            spans = self.get_spans()
+
+            # get SNS messages via SQS
+            response = self.sqs_client.receive_message(
+                QueueUrl=self.sqs_test_queue["QueueUrl"],
+                MessageAttributeNames=["_datadog"],
+                WaitTimeSeconds=2,
+            )
+
+            # clean up resources
+            sns.delete_topic(TopicArn=topic_arn)
+
+            # check if the appropriate span was generated
+            assert spans
+            span = spans[0]
+            assert len(spans) == 2
+            assert span.get_tag("aws.region") == region
+            assert span.get_tag("region") == region
+            assert span.get_tag("aws.operation") == "PublishBatch"
+            assert span.get_tag("params.MessageBody") is None
+            assert_is_measured(span)
+            assert_span_http_status_code(span, 200)
+            assert span.service == "test-botocore-tracing.sns"
+            assert span.resource == "sns.publishbatch"
+            trace_json = span.get_tag("params.MessageAttributes._datadog.StringValue")
+            assert trace_json is None
```

Comment:
Just realized that most of these checks are probably redundant if the snapshot JSON file is the correct expected span output.

Reply:
Yeah, I could remove the `assert`s from these if that makes more sense - I essentially copy/pasted pre-existing tests here

Reply:
Done